### PR TITLE
sync command  support --links

### LINKED ISSUE #1588: Support `--links ` for sync command
From: https://github.com/juicedata/juicesync/issues/121

### PR DESCRIPTION
**Pull request by zhijian-pro**
Close #1588

### CODE REVIEW

File: pkg/object/s3.go [hunk 1]
```diff
@@ -86,10 +86,10 @@ func (s *s3client) Head(key string) (Object, error) {
 		return nil, err
 	}
 	return &obj{
-		key,
```
[the remark this answers is not in the document]
Comment (zhijian-pro) [reply pkg/object/s3.go:89]: ok

File: pkg/object/file.go [hunk 2]
```diff
@@ -326,6 +324,7 @@ func (d *filestore) ListAll(prefix, marker string) (<-chan Object, error) {
 				owner,
 				group,
 				info.Mode(),
+				info.Mode()&os.ModeSymlink != 0,
```
Comment (davies) [inline pkg/object/file.go:327]: In order to follow symlink, we should return the info of it's target (may also ignore the broken symlink?)

File: pkg/object/file.go [hunk 3]
```diff
@@ -281,24 +315,19 @@ func (d *filestore) ListAll(prefix, marker string) (<-chan Object, error) {
...
-					return nil
-				}
 				if os.IsNotExist(err) {
 					logger.Warnf("skip not exist file or directory: %s", path)
 					return nil
 				}
 				listed <- nil
 				logger.Errorf("list %s: %s", path, err)
-				return err
+				return nil
```
Comment (sjp00556) [inline pkg/object/file.go:330]: why not return err?

Comment (sjp00556) [reply pkg/object/file.go:330]: if return nil , sync producer will quit loop,  so here need return err to quit recursive walk

**Pull request merged**